pkg/extensions/cluster_test.go

```diff
@@ -62,15 +72,304 @@ var _ = Describe("Cluster", func() {
 		})
 	})
 
-	Describe("#GetShootStateForCluster", func() {
-		var (
-			expectedShoot      *gardencorev1beta1.Shoot
-			expectedShootState *gardencorev1beta1.ShootState
+	Describe("#SyncClusterResourceToSeed", func() {
+		BeforeEach(func() {
+			expectedCloudProfile = &gardencorev1beta1.CloudProfile{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "core.gardener.cloud/v1beta1",
+					Kind:       "CloudProfile",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+			}
+
+			expectedSeed = &gardencorev1beta1.Seed{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "core.gardener.cloud/v1beta1",
+					Kind:       "Seed",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+			}
+
+			expectedShoot = &gardencorev1beta1.Shoot{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "core.gardener.cloud/v1beta1",
+					Kind:       "Shoot",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: "garden-bar",
+				},
+			}
+
+			clusterName = "shoot--" + expectedShoot.Namespace + "--" + expectedShoot.Name
+			cluster = &extensionsv1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterName,
+				},
+			}
+		})
+
+		It("should not sync cloudprofile, shoot and seed to cluster if seed is not assigned to shoot", func() {
+			Expect(fakeGardenClient.Create(ctx, expectedCloudProfile)).To(Succeed())
+			Expect(fakeGardenClient.Create(ctx, expectedSeed)).To(Succeed())
+			Expect(fakeGardenClient.Create(ctx, expectedShoot)).To(Succeed())
+			Expect(fakeSeedClient.Create(ctx, cluster)).To(Succeed())
```

Comment:
Why do we have to create the cluster resource here? Shouldn't checking that it doesn't exist after calling `SyncClusterResourceToSeed` be also enough?